core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -455,6 +455,25 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
 
   final def compose[R1, E1 >: E](that: ZIO[R1, E1, R]): ZIO[R1, E1, A] = self <<< that
 
+  /**
+   * Taps the effect, printing the result of calling `.toString` on the value
+   */
+  final def debug: ZIO[R, E, A] =
+    self.tapBoth(
```

Comment:
I would how we should handle a defect here? It seems like it could be useful for debugging but the full cause can be extremely long. Maybe we could just show the first cause in that case?

Reply:
Something like?
```scala
  /**
   * Taps the effect, printing the result of calling `.toString` on the value
   * or the error.
   */
  final def debug: ZIO[R, E, A] =
    self
      .tapCause(cause =>
        cause.find {
          case Cause.Interrupt(fiberId) => s"<INTERRUPT> by $fiberId"
          case Cause.Die(throwable)     => s"<DIE> $throwable"
          case Cause.Fail(error)        => s"<FAIL> $error"
        }.fold(UIO.unit) { message =>
          UIO(println(message))
        }
      )
      .tap(value => UIO(println(value)))

  /**
   * Taps the effect, printing the result of calling `.toString` on the value.
   * Prefixes the output with the given message.
   */
  final def debug(prefix: => String): ZIO[R, E, A] =
    self
      .tapCause(cause =>
        cause.find {
          case Cause.Interrupt(fiberId) => s"$prefix: <INTERRUPT> by $fiberId"
          case Cause.Die(throwable)     => s"$prefix: <DIE> $throwable"
          case Cause.Fail(error)        => s"$prefix: <FAIL> $error"
        }.fold(UIO.unit) { message =>
          UIO(println(message))
        }
      )
      .tap(value => UIO(println(s"$prefix: $value")))
```

Reply:
Also, I wouldn't mind not keeping it as it is, and letting the runtime render dies.